cpp/dolfinx/fem/CoordinateElement.cpp

```diff
@@ -86,8 +95,110 @@ void CoordinateElement::compute_reference_geometry(
   assert(K.dimension(1) == this->topological_dimension());
   assert(K.dimension(2) == this->geometric_dimension());
 
-  assert(_compute_reference_geometry);
-  _compute_reference_geometry(X.data(), J.data(), detJ.data(), K.data(),
-                              num_points, x.data(), cell_geometry.data());
+  const int d = cell_geometry.rows();
+  Eigen::Tensor<double, 3, Eigen::RowMajor> phi_tensor(1, 1, d);
+  Eigen::Map<
+      Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>
+      phi(phi_tensor.data(), d, 1);
+
+  Eigen::Tensor<double, 4, Eigen::RowMajor> dphi_tensor(1, 1, d, _tdim);
+  Eigen::Map<
+      Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>
+      dphi(dphi_tensor.data(), d, _tdim);
+
+  if (_is_affine)
+  {
+    Eigen::RowVectorXd X0(_tdim);
+    X0.setZero();
+    Eigen::VectorXd x0(x.cols());
+    // Compute physical coordinates at X=0.
```

Comment:
Are all reference elements parameterized for the interval [-1,1]^d in FIAT? If so, we could add a note that this would be the midpoint of the element, (and maybe add the note from: https://github.com/FEniCS/ffcx/blob/master/ffcx/codegeneration/coordinate_mapping.py#L469)